src/DssDirectDepositAaveDai.sol

```diff
@@ -246,94 +273,139 @@ contract DssDirectDepositAaveDai {
         emit Wind(amount);
     }
 
-    function unwind(uint256 amount, uint256 fees) external {
-        require(wards[msg.sender] == 1 || live == 0, "DssDirectDepositAaveDai/not-authorized");
+    function _unwind(uint256 supplyReduction, uint256 availableLiquidity, Mode mode) internal {
+        // IMPORTANT: this function assumes Vat rate of this ilk will always be == 1 * RAY (no fees).
+        // That's why it converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+        // This module will have an unintended behaviour if rate is changed to some other value.
+
+        bytes32 ilk_ = ilk;
+        address end;
+        uint256 adaiBalance = adai.balanceOf(address(this));
+        uint256 daiDebt;
+        if (mode == Mode.NORMAL) {
+            // Normal mode or module just caged (no culled)
+            // debt is obtained from CDP ink (or art which is the same)
+            (daiDebt,) = vat.urns(ilk_, address(this));
+        } else if (mode == Mode.MODULE_CULLED) {
+            // Module shutdown and culled
+            // debt is obtained from free collateral owned by this contract
+            daiDebt = vat.gem(ilk_, address(this));
+        } else {
+            // MCD caged
+            // debt is obtained from free collateral owned by the End module
+            end = chainlog.getAddress("MCD_END");
+            EndLike(end).skim(ilk_, address(this));
```

Comment:
Wanting to verify `skim()` can be called multiple times. This appears to be the case, but just double checking.

Reply:
Yeah, `skim()` can be called several times, but after the CDP is grabbed, it will be a no op.
For a normal CDP this will happen the first time, in this case it would also happen the first time if the deposit contract isn't culled. In that case the `skim` that will count will be the first one after `uncull` is called.